build-system/tasks/helpers.js

```diff
@@ -223,8 +223,19 @@ function toEsmName(name) {
   return name.replace(/\.js$/, '.mjs');
 }
 
-function maybeToEsmName(name) {
-  return argv.esm ? toEsmName(name) : name;
+function toSxgName(name) {
+  // For certain files during compilation, there seems to be files that get the extension written twice,
+  // e.g. sxg.sxg.js. This is the quickest fix for now.
+  return name.includes('.sxg.js') ? name : name.replace(/\.js$/, '.sxg.js');
+}
+
+function chooseExtension(name) {
+  if (argv.sxg) {
+    return toSxgName(name);
+  } else if (argv.esm) {
+    return toEsmName(name);
+  }
```

Comment:
I suggest deleting `toEsmName` and `toSxgName` and consolidating the logic in this function.

Reply:
Done.